tests/cmdline/commands/test_rabbitmq.py

```diff
@@ -0,0 +1,138 @@
+    monkeypatch.setattr(cmd_rabbitmq, 'get_process_tasks', lambda *args: [1, 2])
+
+    result = run_cli_command(cmd_rabbitmq.cmd_tasks_analyze, raises=True)
+    assert 'There are active processes without process task:' in result.output
+    assert 'Inconsistencies detected between database and RabbitMQ.' in result.output
+
+
+@pytest.mark.usefixtures('stopped_daemon_client')
+def test_tasks_analyze_verbosity(monkeypatch, run_cli_command):
+    """Test the ``tasks analyze`` command with ``-v INFO```."""
+    monkeypatch.setattr(cmd_rabbitmq, 'get_active_processes', lambda *args: [1, 2, 3, 4])
+    monkeypatch.setattr(cmd_rabbitmq, 'get_process_tasks', lambda *args: [1, 2])
+
+    result = run_cli_command(cmd_rabbitmq.cmd_tasks_analyze, ['-v', 'INFO'], raises=True)
+    assert 'Active processes: [1, 2, 3, 4]' in result.output
+    assert 'Process tasks: [1, 2]' in result.output
+
+
+@pytest.mark.usefixtures('stopped_daemon_client')
+def test_tasks_revive_without_daemon(run_cli_command):
+    """Test that ``tasks revive`` fails if no daemon is running."""
+    assert run_cli_command(cmd_rabbitmq.cmd_tasks_revive, raises=True)
+
+
+@pytest.mark.usefixtures('aiida_profile')
+def test_revive(run_cli_command, monkeypatch, aiida_local_code_factory, submit_and_await, started_daemon_client):
+    """Test ``tasks revive``."""
+    code = aiida_local_code_factory('core.arithmetic.add', '/bin/bash')
+    builder = code.get_builder()
+    builder.x = Int(1)
+    builder.y = Int(1)
+    builder.metadata.options.resources = {'num_machines': 1}
+
+    # Temporarily patch the ``RemoteProcessThreadController.continue_process`` method to do nothing and just return a
+    # completed future. This ensures that the submission creates a process node in the database but no task is sent to
+    # RabbitMQ and so the daemon will not start running it.
+    with monkeypatch.context() as context:
+        context.setattr(RemoteProcessThreadController, 'continue_process', lambda *args, **kwargs: None)
+        node = submit(builder)
+
+    # The node should now be created in the database but "stuck"
+    assert node.process_state == ProcessState.CREATED
+
+    # Time to revive it by recreating the task and send it to RabbitMQ
+    run_cli_command(cmd_rabbitmq.cmd_tasks_revive, [str(node.pk), '--force'])
+
+    # Wait for the process to be picked up by the daemon and completed. If there is a problem with the code, this call
+    # should timeout and raise an exception
+    submit_and_await(node)
+    assert node.is_finished_ok
+
+    # Need to manually stop the daemon such that it cleans all state related to the submitted processes. It is not quite
+    # understood how, but leaving the daemon running, a following test that will submit to the daemon may hit an
+    # an exception because the node submitted here would no longer exist and the daemon would try to operate on it.
+    started_daemon_client.stop_daemon(wait=True)
```

Comment:
If understand correctly, this stops the daemon the second time since `stop_daemon` is already called from `daemon_client` fixture of `started_daemon_client`?

Reply:
Hmm, I think it might indeed no longer be necessary. When I introduced the `daemon_client` fixture, it was originally session-scoped, just for efficiency to only stop it once at the end of the session. But I think exactly for this reason I decided to make it function scoped because it can cause side effects between tests. I can go back and remove it and see if the tests run. They probably will now.